node/src/BaseClient.ts

```diff
@@ -278,7 +286,57 @@ export class BaseClient {
     private remainingReadData: Uint8Array | undefined;
     private readonly requestTimeout: number; // Timeout in milliseconds
     private isClosed = false;
+    private readonly pubsubFutures: [PromiseFunction, ErrorFunction][] = [];
+    private pendingPushNotification: response.Response[] = [];
+    private config: BaseClientConfiguration | undefined;
+
+    protected configurePubsub(
+        options: ClusterClientConfiguration | GlideClientConfiguration,
+        configuration: connection_request.IConnectionRequest,
+    ) {
+        if (options.pubsubSubscriptions) {
+            if (options.protocol == ProtocolVersion.RESP2) {
+                throw new ConfigurationError(
+                    "PubSub subscriptions require RESP3 protocol, but RESP2 was configured.",
+                );
+            }
+
+            const { context, callback } = options.pubsubSubscriptions;
+
+            if (context && !callback) {
+                throw new ConfigurationError(
+                    "PubSub subscriptions with a context require a callback function to be configured.",
+                );
+            }
+
+            // Initialize pubsubSubscriptions in the configuration if not already present
+            if (!configuration.pubsubSubscriptions) {
```

Comment:
why not to always initialize?

Reply:
I wasnt sure if this is true since in the python code it looks like it was already initialized, i can do that if you prefer so

Reply:
it looks odd, the pubsubSubscriptions either exist in the constructor or not

Reply:
#1974

Reply:
done